itlwm/hal_iwn/ItlIwn.cpp

```diff
@@ -495,7 +495,8 @@ iwn_attach(struct iwn_softc *sc, struct pci_attach_args *pa)
         IEEE80211_C_MONITOR |    /* monitor mode supported */
         IEEE80211_C_SHSLOT |    /* short slot time supported */
         IEEE80211_C_SHPREAMBLE |    /* short preamble supported */
-        IEEE80211_C_PMGT;        /* power saving supported */
+        IEEE80211_C_PMGT |        /* power saving supported */
+        IEEE80211_C_MFP;    /* management frame protection 11w supported */
```

Comment:
Not all of the DVM cards' firmwares support 11w, original `iwlwifi` code says:
```
/*
 * Enable 11w if advertised by firmware and software crypto
 * is not enabled (as the firmware will interpret some mgmt
 * packets, so enabling it with software crypto isn't safe)
 */
if (priv->fw->ucode_capa.flags & IWL_UCODE_TLV_FLAGS_MFP &&
    !iwlwifi_mod_params.swcrypto)
	ieee80211_hw_set(hw, MFP_CAPABLE);
```

Reply:
You made a good point.

I quickly checked my N 6205 card, the tlv_feature_flags is 0x0b, and it doesn't include the bit IWN_UCODE_TLV_FLAGS_MFP (1<<2). It looks like the firmware doesn't tell support 11w, and it is actually working on my setup. It is unclear to me what's the best way to determine this feature flag.

I'll leave the code for iwn as is without testing the firmware feature flags for now. May be there are some corner cases to be discovered.

Reply:
I think I know why the dvm firmware doesn't support 11w. The firmware doesn't support install igtk hw keys.
In the current implementation, group key and the new igtk are done in software crypto in the net80211 stack for both `iwn` and `iwm`. Unicast key is handled in firmware. With this hybrid approach, we can consider the driver is MFP capable.